internal/proxyservice/http/basicauth/connector.go

```diff
@@ -0,0 +1,63 @@
+package basicauth
+
+import (
+	"encoding/base64"
+	"fmt"
+	gohttp "net/http"
+	"strconv"
+	"strings"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/log"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+)
+
+// Connector injects an HTTP request with AWS authorization headers.
+type Connector struct {
+	logger log.Logger
+}
+
+// Connect implements the http.Connector func signature.
+func (c *Connector) Connect(
+	r *gohttp.Request,
+	credentialsByID connector.CredentialValuesByID,
+) error {
+	var ok bool
+
+	// Validate credentials
```

Comment:
Just a comment for the future. We definitely should be checking this long before the service connectors start. I suppose maybe here too, because I guess a value could be specified in the Secretless config but the secret provider could have an empty value...

Reply:
Yeah we need the runtime checks here because this is validating "that the values we need from  the provider exist at this moment".  You're saying there should separately be validation of the config at the time it's loaded so that if you specify connector with id X that requires fields A, B, C, then we validate that A, B, and C exist in the creds section.

I agree this would be a good feature, but it's definitely a new feature and isn't trivial to design.  That is, it would impact the connector interface: you'd need a way to specify required fields, either in PluginInfo or via a new method we'd require.  Given that the added benefit over the runtime checks above is failing faster (ie, you catch your bad config at startup rather than runtime), I'm actually not sure it's worth the added complexity, but it might be.

Reply:
This is definitely one for later. I agree it'll require some design.